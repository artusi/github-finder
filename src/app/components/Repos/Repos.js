import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { requestSortUpdate } from "store/state/github";
import { RepoCard, OptionSelect } from "ui";
import styles from "./Repos.module.css";

// eslint-disable-next-line no-shadow
export function Repos({ github, requestSortUpdate }) {
  const { error, username, repos } = github;
  const { byId, all } = repos;
  const errorMessage = "Sorry but we couldn't find this user ;-(";

  if (error) return <div className={styles.repos}>{errorMessage}</div>;

  const sortOptions = [
    { value: "full_name", label: "Name" },
    { value: "updated", label: "Updated" },
    { value: "created", label: "Created" }
  ];

  return (
    <div className={styles.repos}>
      <div className={styles.disclamer}>
        <div className={styles.info}>
          Hey, {username} has {repos.total} repositories.
        </div>
        <div className={styles.sorter}>
          Sort by:
          <OptionSelect
            id="sortRepos"
            options={sortOptions}
            onChange={value => {
              requestSortUpdate(value);
            }}
          />
        </div>
      </div>
      <div className={styles.reposList}>
        {all.map(id => {
          const currentRepo = byId[id];
          const title = currentRepo.name;
          return (
            <div key={currentRepo.id} className={styles.reposListItem}>
              <RepoCard
                {...currentRepo}
                title={title}
                linkFinderPath={`${username}/${title}`}
              />
            </div>
          );
        })}
      </div>
    </div>
  );
}

const mapStateToProps = state => ({
  github: state.github
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      requestSortUpdate
    },
    dispatch
  );
}

Repos.propTypes = {
  requestSortUpdate: PropTypes.func.isRequired,
  github: PropTypes.shape({
    error: PropTypes.bool.isRequired,
    username: PropTypes.string.isRequired,
    repos: PropTypes.object.isRequired
  }).isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Repos);
