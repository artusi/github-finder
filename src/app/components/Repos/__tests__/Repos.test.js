import React from "react";
import { Repos } from "../Repos";

describe("<Repos />", () => {
  const initialProps = {
    github: {
      error: false,
      username: "Vegeta",
      repos: {
        total: 2,
        all: ["4521396", "58581481"],
        byId: {
          "4521396": {
            id: "4521396",
            name: "jclip d asdnlkasdkas dlaks dlkas dlkas d lkda",
            description:
              'JClip is a new way to make animations and control jpegs, you can use as a "gif" file or control like a Flash Movieclip.',
            language: "JavaScript",
            stars: 1
          },
          "58581481": {
            id: "58581481",
            name: "git-alias",
            description: "Some tests with alias",
            language: "Not declared",
            stars: 0
          }
        }
      }
    }
  };

  it("renders", () => {
    const wrapper = shallow(<Repos {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders > with initial user", () => {
    const propsWithError = { ...initialProps };
    propsWithError.github.error = true;

    const wrapper = shallow(<Repos {...propsWithError} />);

    expect(wrapper).toMatchSnapshot();
  });
});
