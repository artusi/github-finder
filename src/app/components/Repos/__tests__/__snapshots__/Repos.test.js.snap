// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repos /> renders > with initial user 1`] = `
<div
  className="repos"
>
  Sorry but we couldn't find this user ;-(
</div>
`;

exports[`<Repos /> renders 1`] = `
<div
  className="repos"
>
  <div
    className="disclamer"
  >
    <div
      className="info"
    >
      Hey, 
      Vegeta
       has 
      2
       repositories.
    </div>
    <div
      className="sorter"
    >
      Sort by:
      <OptionSelect
        customClassName=""
        disabled={false}
        id="sortRepos"
        initialValue=""
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Name",
              "value": "name",
            },
            Object {
              "label": "Updated",
              "value": "updated",
            },
            Object {
              "label": "Stars",
              "value": "stars",
            },
          ]
        }
      />
    </div>
  </div>
  <div
    className="reposList"
  >
    <div
      className="reposListItem"
      key="4521396"
    >
      <RepoCard
        customClassName=""
        description="JClip is a new way to make animations and control jpegs, you can use as a \\"gif\\" file or control like a Flash Movieclip."
        id="4521396"
        language="JavaScript"
        linkFinderPath="Vegeta/jclip d asdnlkasdkas dlaks dlkas dlkas d lkda"
        name="jclip d asdnlkasdkas dlaks dlkas dlkas d lkda"
        stars={1}
        title="jclip d asdnlkasdkas dlaks dlkas dlkas d lkda"
      />
    </div>
    <div
      className="reposListItem"
      key="58581481"
    >
      <RepoCard
        customClassName=""
        description="Some tests with alias"
        id="58581481"
        language="Not declared"
        linkFinderPath="Vegeta/git-alias"
        name="git-alias"
        stars={0}
        title="git-alias"
      />
    </div>
  </div>
</div>
`;
